x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/SecurityRestFilter.java

```diff
@@ -108,47 +87,26 @@ public void handleRequest(RestRequest request, RestChannel channel, NodeClient c
                 if (secondaryAuthentication != null) {
                     logger.trace("Found secondary authentication {} in REST request [{}]", secondaryAuthentication, request.uri());
                 }
-                try {
-                    doHandleRequest(request, channel, client);
-                } catch (Exception e) {
-                    handleException(ActionType.RequestHandling, request, channel, e);
-                }
-            }, e -> handleException(ActionType.SecondaryAuthentication, request, channel, e)));
-        }, e -> handleException(ActionType.Authentication, request, channel, e)));
+                doHandleRequest(request, channel, client);
+            }, e -> handleException(request, channel, e)));
+        }, e -> handleException(request, channel, e)));
     }
 
     private void doHandleRequest(RestRequest request, RestChannel channel, NodeClient client) throws Exception {
         threadContext.sanitizeHeaders();
-        restHandler.handleRequest(request, channel, client);
+        try {
+            restHandler.handleRequest(request, channel, client);
+        } catch (Exception e) {
+            logger.debug(() -> format("Request handling failed for REST request [%s]", request.uri()), e);
+            throw e;
+        }
     }
 
-    protected void handleException(ActionType actionType, RestRequest request, RestChannel channel, Exception e) {
-        logger.debug(() -> format("%s failed for REST request [%s]", actionType, request.uri()), e);
+    protected void handleException(RestRequest request, RestChannel channel, Exception e) {
+        logger.debug(() -> format("failed for REST request [%s]", request.uri()), e);
         threadContext.sanitizeHeaders();
-        final RestStatus restStatus = ExceptionsHelper.status(e);
         try {
-            channel.sendResponse(new RestResponse(channel, restStatus, e) {
-
-                @Override
-                protected boolean skipStackTrace() {
-                    return restStatus == RestStatus.UNAUTHORIZED;
-                }
-
-                @Override
-                public Map<String, List<String>> filterHeaders(Map<String, List<String>> headers) {
-                    if (actionType != ActionType.RequestHandling
-                        || (restStatus == RestStatus.UNAUTHORIZED || restStatus == RestStatus.FORBIDDEN)) {
-                        if (headers.containsKey("Warning")) {
-                            headers = Maps.copyMapWithRemovedEntry(headers, "Warning");
-                        }
-                        if (headers.containsKey("X-elastic-product")) {
-                            headers = Maps.copyMapWithRemovedEntry(headers, "X-elastic-product");
-                        }
-                    }
-                    return headers;
-                }
-
-            });
+            channel.sendResponse(new RestResponse(channel, e));
```

Comment:
This here is the goal: `new RestResponse(channel, e)` is how responses are generated from exceptions everywhere else in the codebase.
This is important because authN failure exceptions will be signaled like exceptions for invalid/malformed requests.